src/components/Dashboard.jsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+
+function Dashboard() {
+  const [listNameArr, setListsArr] = React.useState([]);
+  //const [bestSellersArr, setbestSellersArr] = React.useState([]);
+  const [rankMovementArr, setRankMoveArr] = React.useState([]);
+  const [weeksOnListArr, setWeeksOnListArr] = React.useState([]);
+  const [numResults, setNumResults] = React.useState([]);
+  const [publisherDist, setPublisherDist] = React.useState([]);
+  const [isbn, setISBNArr] = React.useState([]);
+  //const apikey = ${CS464P-564_Final_Project.env.NYT_API_KEY};
+
+  React.useEffect(() => {
+    fetch(
+      "https://api.nytimes.com/svc/books/v3/lists/full-overview.json?api-key=6P7g1cNgyA4yxbQWfxkMkq3hZi8RXYZp"
+    )
+      .then((response) => response.json())
+      .then((data) => routeAndParseData(data.results.lists));
+  }, []);
+
+  function routeAndParseData(data) {
+    setListsArr(listObjectToArr(data));
+    setWeeksOnListArr(createWeeksOnListArray(data));
+    setRankMoveArr(createRankMovement(data));
+    setPublisherDist(createPublisherDist(data));
+    setISBNArr(createISBNArr(data));
+  }
+
+  function createRankMovement(data) {
+    const key = ["new", "noChange", "up", "down"];
+    let rankValues = [0, 0, 0, 0];
+    data.forEach((list) => {
+      list.books.forEach((book) => {
+        if (book.rank_last_week === 0) {
+          rankValues[0]++;
+        } else if (book.rank_last_week === book.rank) {
+          rankValues[1]++;
+        } else if (book.rank > book.rank_last_week) {
+          rankValues[2]++;
+        } else {
+          rankValues[3]++;
+        }
+      });
+    });
+    console.log(rankValues);
+    return rankValues;
+  }
+
+  //Array with all of the category names. We can use this to map all of the other arrays later
+  function listObjectToArr(data) {
+    let listNames = [];
+    data.forEach((list) => {
+      listNames.push(list.list_name);
+    });
+    console.log(listNames);
+    return listNames;
+  }
+
+  //Creates array of all the "weeks on list" data points here we have one big array and one arrray of arrays divided by cateogry depending on
+  function createWeeksOnListArray(data) {
+    let weeksOnList = [];
+    let weeksByCat = [];
+    let temp = [];
+    data.forEach((list) => {
+      list.books.forEach((book) => {
+        weeksOnList.push(book.weeks_on_list);
+        temp.push(book.weeks_on_list);
+      });
+      weeksByCat.push(temp);
```

Comment:
To make sure that these arrays stay connected to the correct list value, maybe this should be a dictionary instead. Use the list title as the key and the associated array as the value itself?

Reply:
This could be helpful for any other 2d arrays based on list as well, such as the publishers section.